static/src/js/components/AccessMethod/AccessMethod.jsx

```diff
@@ -57,260 +41,105 @@ const EchoForm = lazy(() => import('./EchoForm'))
+const AccessMethod = ({
+  accessMethods,
+  index,
+  isActive,
+  metadata,
+  onSelectAccessMethod,
+  onSetActivePanel,
+  onTogglePanels,
+  onUpdateAccessMethod,
+  selectedAccessMethod,
+  shapefileId,
+  spatial,
+  temporal,
+  ursProfile,
+  granuleMetadata,
+  projectCollection
+}) => {
+  const { [selectedAccessMethod]: selectedMethod = {} } = accessMethods
+
+  const {
+    form,
+    rawModel = null,
+    selectedVariables = [],
+    selectedOutputFormat,
+    selectedOutputProjection,
+    supportedOutputFormats = [],
+    supportedOutputProjections = [],
+    supportsTemporalSubsetting = false,
+    supportsShapefileSubsetting = false,
+    supportsBoundingBoxSubsetting = false,
+    supportsVariableSubsetting = false,
+    supportsConcatenation = false,
+    supportsSwodlr = false,
+    defaultConcatenation = false,
+    enableTemporalSubsetting: hasEnabledTemporalSubsetting
+  } = selectedMethod || {}
+
+  const { isRecurring } = temporal
+
+  // EnabledTemporalSubsetting by default
+  let setTemporal = true
+
+  // If enabledTemporalSubsetting is explictly false, set the initial value to false
+  if (hasEnabledTemporalSubsetting === false) {
+    setTemporal = false
   }
 
-  UNSAFE_componentWillReceiveProps() {
-    const { temporal, granuleMetadata, projectCollection } = this.props
-
-    const { isRecurring } = temporal
-
-    // Disable temporal subsetting if the user has a recurring date selected
-    if (isRecurring) {
-      this.setState({
-        enableTemporalSubsetting: false
-      })
-    }
-
-    const {
-      granules: projectCollectionGranules = {}
-    } = projectCollection
-
-    const {
-      addedGranuleIds = [],
-      allIds: granulesAllIds = []
-    } = projectCollectionGranules
-
-    let granulesToDisplay = []
+  // Initialize State Variables
+  const [enableTemporalSubsetting, setEnableTemporalSubsetting] = useState(setTemporal)
+  const [selectedHarmonyMethodName, setSelectedHarmonyMethodName] = useState('')
+  const [isHarmony, setIsHarmony] = useState(selectedAccessMethod.startsWith('harmony'))
+  const [enableSpatialSubsetting, setEnableSpatialSubsetting] = useState(false)
+  const [enableConcatenateDownload, setEnableConcatenateDownload] = useState(defaultConcatenation)
+  const [granuleList, setGranuleList] = useState([])
+  // Only set state on re-render
+
+  const {
+    granules: projectCollectionGranules = {}
+  } = projectCollection
+
+  const {
+    addedGranuleIds = [],
+    allIds: granulesAllIds = []
+  } = projectCollectionGranules
+
+  let granulesToDisplay = []
+
+  if (addedGranuleIds.length > 0) {
+    granulesToDisplay = addedGranuleIds
+  } else {
+    granulesToDisplay = granulesAllIds
+  }
 
-    if (addedGranuleIds.length > 0) {
-      granulesToDisplay = addedGranuleIds
-    } else {
-      granulesToDisplay = granulesAllIds.slice(0, 10)
-    }
+  const granuleListObj = []
 
-    const granuleList = []
+  granulesToDisplay.forEach((id) => {
```

Comment:
This should be a `.map` instead of looking through an array and pushing to a different array